MekHQ/src/mekhq/campaign/universe/PlanetarySystem.java

```diff
@@ -169,28 +170,27 @@ public String getName(LocalDate when) {
     }
 
     public List<String> getFactions(LocalDate when) {
-        List<String> factions = new ArrayList<>();
-        for (Planet planet : planets.values()) {
-            List<String> f = planet.getFactions(when);
-            if (null != f) {
-                factions.addAll(f);
-            }
-        }
+        List<String> factions = planets.values().stream()
+                .map(planet -> planet.getFactions(when))
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+
         return factions;
     }
 
     public Set<Faction> getFactionSet(LocalDate when) {
-        Set<Faction> factions = new HashSet<>();
-        for (Planet planet : planets.values()) {
-            Set<Faction> f = planet.getFactionSet(when);
-            if (null != f) {
-                factions.addAll(f);
-            }
-        }
+        Set<Faction> factions = planets.values().stream()
```

Comment:
I'm kind of curious to see a "how long does this take to run" comparison between this and the older code. In my experience, the syntactic sugar carries a performance penalty, but I'm happy to be proven wrong.

Reply:
I was under the impression that streams were faster than loops, but that appears to not be as universally true as I was led to believe.

According to google, streams are faster when looping through large volumes of data, but slower with small volumes. This is definitely a case of 'small volume' so I'll go ahead and revert this change and be more judicious with their usage in future.